internal/project/repository/eventsourcing/model.go

```
@@ -98,3 +156,51 @@ func (p *Project) appendReactivatedEvent() error {
 	p.State = model.ProjectStateToInt(model.Active)
 	return nil
 }
+
+func (p *Project) appendAddMemberEvent(event *es_models.Event) error {
+	member, err := getMemberData(event)
+	if err != nil {
+		return err
+	}
+	member.ObjectRoot.CreationDate = event.CreationDate
+	p.Members = append(p.Members, member)
+	return nil
+}
+
+func (p *Project) appendChangeMemberEvent(event *es_models.Event) error {
+	member, err := getMemberData(event)
+	if err != nil {
+		return err
+	}
+	for i, m := range p.Members {
+		if m.UserID == member.UserID {
+			p.Members[i] = member
+		}
+	}
+	return nil
+}
+
+func (p *Project) appendRemoveMemberEvent(event *es_models.Event) error {
+	member, err := getMemberData(event)
+	if err != nil {
+		return err
+	}
+	for i, m := range p.Members {
+		if m.UserID == member.UserID {
+			p.Members[i] = p.Members[len(p.Members)-1]
+			p.Members[len(p.Members)-1] = nil
+			p.Members = p.Members[:len(p.Members)-1]
+		}
+	}
+	return nil
+}
+
+func getMemberData(event *es_models.Event) (*ProjectMember, error) {
+	member := &ProjectMember{}
+	member.ObjectRoot.AppendEvent(event)
+	if err := json.Unmarshal(event.Data, member); err != nil {
+		logging.Log("EVEN-e4dkp").WithError(err).Error("could not unmarshal event data")
+		return nil, err
```

i would return a caos error here